velox/exec/ArrowStream.cpp

```diff
@@ -0,0 +1,76 @@
+/*
+ * Copyright (c) Facebook, Inc. and its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#include "velox/exec/ArrowStream.h"
+
+namespace facebook::velox::exec {
+
+ArrowStream::ArrowStream(
+    int32_t operatorId,
+    DriverCtx* driverCtx,
+    const std::shared_ptr<const core::ArrowStreamNode>& arrowStream)
+    : SourceOperator(
+          driverCtx,
+          arrowStream->outputType(),
+          operatorId,
+          arrowStream->id(),
+          "Arrow Stream") {
+  pool_ = arrowStream->memoryPool();
+  arrowStream_ = arrowStream->arrowStream();
+
+  // Get arrow schema.
+  if (arrowStream_->get_schema(arrowStream_.get(), &arrowSchema_)) {
+    VELOX_FAIL(
+        "Failed to call get_schema on ArrowStream: " + std::string(GetError()))
+  }
+}
+
+ArrowStream::~ArrowStream() {
+  if (!isFinished()) {
+    close();
+  }
+}
+
+RowVectorPtr ArrowStream::getOutput() {
+  struct ArrowArray arrowArray;
+  if (arrowStream_->get_next(arrowStream_.get(), &arrowArray)) {
+    VELOX_FAIL(
+        "Failed to call get_next on ArrowStream: " + std::string(GetError()))
+  }
+  if (arrowArray.release == NULL) {
```

Comment:
What is NULL? Can you use nullptr instead?

Reply:
Arrow marks it as NULL in release callback, so we follow that here. 
https://github.com/apache/arrow/blob/master/cpp/src/arrow/c/helpers.h#L63-L64